sklearn_numba_dpex/kmeans/kernels/lloyd_single_step.py

```diff
@@ -106,22 +106,46 @@ def make_lloyd_single_step_fixed_window_kernel(
     last_centroid_window_idx = n_windows_for_centroids - 1
     last_feature_window_idx = n_windows_for_features - 1
 
-    centroids_window_shape = (centroids_window_height, centroids_window_width)
-
-    global_size = (math.ceil(n_samples / work_group_size)) * (work_group_size)
-    n_subgroups = global_size // sub_group_size
+    n_subgroups = math.ceil(n_samples / window_n_centroids)
+    global_size = (
+        math.ceil(n_subgroups / centroids_window_height) * centroids_window_height,
+        window_n_centroids,
+    )
 
     n_cluster_items = n_clusters * (n_features + 1)
     n_cluster_bytes = np.dtype(dtype).itemsize * n_cluster_items
     # TODO: control that this value is not higher than the number of sub-groups of size
     # sub_group_size that can effectively run concurrently. We should fetch this
     # information and apply it here.
+
+    # NB: for more details about the privatization strategy the following variables
+    # refer too, please read the inline commenting that address it with more details in
+    # the kernel definition.
```

Comment:
```suggestion
    # refer too, please read the inline comments in the kernel definition.
```

to avoid duplicating "more details".